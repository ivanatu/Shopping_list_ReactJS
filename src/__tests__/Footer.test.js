import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import renderer from 'react-test-renderer'
import Footer from '../components/footer'
import '../setupTests';

describe('<Footer/>', () => {
    const wrapper = shallow( <Footer/> );
    it('has a valid snapshot', () => {
        const component = renderer.create(
        <Footer />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing', () => {
        shallow(<Footer/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div/>))
    });
    it('should render <footer> tag without throwing an error', () => {
        expect(wrapper.exists(<footer className="fixed-bottom"/>))
    });
    it('should render <div> tag without throwing an error', () => {
        expect(wrapper.exists(<div className="container"/>))
    });
    it('should render <span> tag without throwing an error', () => {
        expect(wrapper.exists(<span className="navbar-text ml-auto"/>))
    });
});