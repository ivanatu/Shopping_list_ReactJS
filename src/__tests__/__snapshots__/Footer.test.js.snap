// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> has a valid snapshot 1`] = `
<div>
  <footer
    className="fixed-bottom"
  >
    <div
      className="container"
    >
      <span
        className="navbar-text ml-auto"
      >
        Â© Shopping List Inc All rights reserved
      </span>
    </div>
  </footer>
</div>
`;
